app/assets/javascripts/Components/assignment_summary_table.jsx

```diff
@@ -29,96 +38,127 @@ class AssignmentSummaryTable extends React.Component {
         col["filterable"] = false;
         col["defaultSortDesc"] = true;
       });
+      const markingStates = this.getMarkingStates(res.data);
       this.setState({
         data: res.data,
         criteriaColumns: res.criteriaColumns,
         num_assigned: res.numAssigned,
         num_marked: res.numMarked,
         loading: false,
+        marking_states: markingStates,
       });
     });
   };
 
-  fixedColumns = [
-    {
-      Header: I18n.t("activerecord.models.group.one"),
-      id: "group_name",
-      accessor: "group_name",
-      Cell: row => {
-        if (row.original.result_id) {
-          const path = Routes.edit_assignment_submission_result_path(
-            this.props.assignment_id,
-            row.original.submission_id,
-            row.original.result_id
-          );
-          return <a href={path}>{row.original.group_name}</a>;
-        } else {
-          return <span>{row.original.group_name}</span>;
-        }
-      },
-      filterMethod: (filter, row) => {
-        if (filter.value) {
-          // Check group name
-          if (row._original.group_name.includes(filter.value)) {
-            return true;
+  onFilteredChange = () => {
+    const summaryTable = this.getWrappedInstance();
+    const markingStates = this.getMarkingStates(summaryTable.state.sortedData);
+    this.setState({marking_states: markingStates});
+  };
+
+  getMarkingStates(data) {
+    const markingStates = {
+      not_collected: 0,
+      incomplete: 0,
+      complete: 0,
+      released: 0,
+      remark: 0,
+      before_due_date: 0,
+    };
+    data.forEach(row => {
+      markingStates[row["marking_state"]] += 1;
+    });
+    return markingStates;
+  }
+
+  getWrappedInstance() {
```

Comment:
Instead of redefining this here, you can just change [line 54](https://github.com/MarkUsProject/Markus/pull/5518/files#diff-e4273cac2d46f2e394362d102b79e399bb43095a55e25ea151767b064ceb15c2R54) to:

```
    const summaryTable = this.wrappedInstance;
```